optimizely/helpers/audience.py

```diff
@@ -50,17 +29,34 @@ def is_user_in_experiment(config, experiment, attributes):
   """
 
   # Return True in case there are no audiences
-  if not experiment.audienceIds:
+  audience_conditions = experiment.getAudienceConditionsOrIds()
+  if not audience_conditions:
     return True
 
   if attributes is None:
     attributes = {}
 
-  # Return True if conditions for any one audience are met
-  for audience_id in experiment.audienceIds:
-    audience = config.get_audience(audience_id)
+  def evaluate_custom_attr(audienceIdToConditionIndexDict):
+    audienceId = list(audienceIdToConditionIndexDict.keys())[0]
+    index = audienceIdToConditionIndexDict[audienceId]
+
+    audience = config.get_audience(audienceId)
+    custom_attr_condition_evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      audience.conditionList, attributes)
+
+    return custom_attr_condition_evaluator.evaluate(index)
 
-    if is_match(audience, attributes):
-      return True
+  def evaluate_audience(audienceId):
+    audience = config.get_audience(audienceId)
```

Comment:
What if the audience can't be found? I think we should gracefully return null instead of throwing an AttributeError when we dereference `audience.conditionStructure`.

My apologies - I didn't think about this until now, so it wasn't in the spec 😬 I'll add a compatibility test for this! It looks like Java and JS are already compliant (although they differ in whether they invoke the error handler; I'd say we shouldn't).